qiskit/circuit/gate.py

```diff
@@ -24,27 +26,67 @@
 class Gate(Instruction):
     """Unitary gate."""
 
-    def __init__(self, name, num_qubits, params, label=None):
+    def __init__(self, name, num_qubits, params, phase_angle=0, label=None):
         """Create a new gate.
 
         Args:
             name (str): the Qobj name of the gate
             num_qubits (int): the number of qubits the gate acts on.
             params (list): a list of parameters.
-            label (str or None): An optional label for the gate [Default: None]
+            phase_angle (float): set the exponent of the gate phase (Default: 0).
+            label (str or None): An optional label for the gate (Default: None).
         """
         self._label = label
         self.definition = None
+        self._phase_angle = 0
+        if phase_angle:
+            self.phase_angle = phase_angle
         super().__init__(name, num_qubits, 0, params)
 
+    @property
+    def phase(self):
+        """Return the phase of the gate."""
+        return cmath.exp(1j * self.phase_angle) if self.phase_angle else 1
```

Comment:
suggest using the attribute directly within the class itself. the setter/getter are external APIs.
```suggestion
        return cmath.exp(1j * self._phase_angle) if self._phase_angle else 1
```
also why the if/else? are we allowing this attribute to be None? the `__init__` default seems to be that we always assume `0` unless otherwise stated, which makes sense to me. None should not really be an option.